main.py

```diff
@@ -112,105 +147,135 @@ def main(args: argparse.Namespace) -> None:
-                    file_path = os.path.join(NUSCENES_ROOT, sensor_data['filename'])
-                    if sensor == 'LIDAR_TOP':
+                    sensor_data = nusc.get("sample_data", sensor_token)
+                    file_path = os.path.join(NUSCENES_ROOT, sensor_data["filename"])
+                    if sensor == "LIDAR_TOP":
                         # nuscenes lidar data is stored as (x, y, z, intensity, ring index)
                         scan = np.fromfile(file_path, dtype=np.float32)
                         points = scan.reshape((-1, 5))
 
                         # Transform lidar points from point sensor frame to egovehicle frame
-                        calibration = nusc.get('calibrated_sensor', sensor_data['calibrated_sensor_token'])
-                        egovehicle_R_lidar = quat2rotmat(calibration['rotation'])
-                        egovehicle_t_lidar = np.array(calibration['translation'])
-                        egovehicle_SE3_lidar = SE3(rotation=egovehicle_R_lidar, translation=egovehicle_t_lidar)
-                        points_egovehicle = egovehicle_SE3_lidar.transform_point_cloud(points[:, :3])
-
-                        data = {"x": points_egovehicle[:, 0], "y": points_egovehicle[:, 1], "z": points_egovehicle[:, 2], "intensity": points[:,3]}
-                        cloud = PyntCloud(pd.DataFrame(data))
-                        cloud_fpath = os.path.join(output_sensor_path, f"PC_{timestamp}.ply")
-                        cloud.to_file(cloud_fpath)
-                    else: 
-                        shutil.copy(file_path, os.path.join(output_sensor_path, f"{argo_sensor}_{timestamp}.jpg"))
+                        calibration = nusc.get(
+                            "calibrated_sensor", sensor_data["calibrated_sensor_token"]
+                        )
+                        egovehicle_R_lidar = quat2rotmat(calibration["rotation"])
+                        egovehicle_t_lidar = np.array(calibration["translation"])
+                        egovehicle_SE3_lidar = SE3(
+                            rotation=egovehicle_R_lidar, translation=egovehicle_t_lidar
+                        )
+                        points_egovehicle = egovehicle_SE3_lidar.transform_point_cloud(
+                            points[:, :3]
+                        )
+
+                        extract_pc(
+                            points_egovehicle, points, output_sensor_path, timestamp
+                        )
+                    else:
+                        shutil.copy(
+                            file_path,
+                            os.path.join(
+                                output_sensor_path, f"{argo_sensor}_{timestamp}.jpg"
+                            ),
+                        )
 
                     if ego_pose is None:
-                        ego_pose = nusc.get('ego_pose', sensor_data['ego_pose_token'])
+                        ego_pose = nusc.get("ego_pose", sensor_data["ego_pose_token"])
```

Comment:
is this the pose for the LiDAR sweep timestamp or for an image timestamp?

Reply:
We loop over all sensors in sample.items() and use timestamp from whichever sensor we get first. I believe it should always be LiDAR, but for samples it probably does not matter since LiDAR and cam have pretty close timestamps.